staging/src/k8s.io/kube-aggregator/pkg/cmd/server/start.go

```diff
@@ -101,7 +101,7 @@ func (o *AggregatorOptions) Complete() error {
 	return nil
 }
 
-func (o AggregatorOptions) RunAggregator(stopCh <-chan struct{}) error {
+func (o *AggregatorOptions) RunAggregator(stopCh <-chan struct{}) error {
 	// TODO have a "real" external address
 	if err := o.RecommendedOptions.SecureServing.MaybeDefaultWithSelfSignedCerts("localhost", nil, nil); err != nil {
```

Comment:
does MaybeDefaultWithSelfSignedCerts mutate the options?

Reply:
Yes, but does not matter. Because `RecommendedOptions` is pointer type, the original mutate also influence AggregatorOptions.

Reply:
The `Config()` func above should not mutate. Neither `RunAggregator` nor `Config()` should mutate their options. I would be fine with an explicit copy of the options though.

Reply:
Same as above, explicit copy of the options does not prevent mutation.

Reply:
value copy can only protect `RecommendedOptions` overwrite by another address. And obviously this not happen here. So it's safe as before.